x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/session/Results.java

```diff
@@ -8,28 +8,76 @@
 
 import org.apache.lucene.search.TotalHits;
 import org.apache.lucene.search.TotalHits.Relation;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.xpack.eql.action.EqlSearchResponse.Count;
+import org.elasticsearch.xpack.eql.action.EqlSearchResponse.Sequence;
 
 import java.util.List;
 
 import static java.util.Collections.emptyList;
 
 public class Results {
 
-    public static final Results EMPTY = new Results(new TotalHits(0, Relation.EQUAL_TO), emptyList());
+    private enum Type {
+        SEARCH_HIT,
+        SEQUENCE,
+        COUNT;
+    }
+    
+    public static final Results EMPTY = new Results(new TotalHits(0, Relation.EQUAL_TO), TimeValue.MINUS_ONE, false, emptyList());
 
     private final TotalHits totalHits;
-    private final List<Object> results;
+    private final List<?> results;
+    private final boolean timedOut;
+    private final TimeValue time;
```

Comment:
Is `time` the amount of time it took to return the results (judging by the `execTime()` method) ? If so, wouldn't be more descriptive to have this named `took` or `duration` or `execTime`?